toolchains/xslt-M4/document/xml/element-reference-html.xsl

```diff
@@ -49,19 +49,32 @@
          <xsl:apply-templates/>
        </div>
    </xsl:template>
-   
+   <!-- The Ancor-Template Generation that supposedly slows down the page response (for ALL MODLES especially) -->
    <xsl:template match="*[exists(@gi)]" expand-text="true">
       <xsl:variable name="level" select="count(ancestor-or-self::*[exists(@gi)])"/>
       <xsl:variable name="header-tag" select="if ($level le 6) then ('h' || $level) else 'p'"/>
       <div class="model-entry definition { tokenize(@_metaschema-xml-id,'/')[2] }">
          <xsl:variable name="header-class" expand-text="true">{ if (exists(parent::map)) then 'definition' else 'instance' }-header</xsl:variable>
          <div class="{ $header-class }">
-            <!-- generates h1-hx headers picked up by Hugo toc -->
-            <xsl:element expand-text="true" name="{ $header-tag }" namespace="http://www.w3.org/1999/xhtml">
-               <xsl:attribute name="id" select="@_tree-xml-id"/>
-               <xsl:attribute name="class">toc{ $level} name</xsl:attribute>
-               <xsl:text>{ @gi }</xsl:text>
+
+            <!-- generates h1-hx headers picked up by Hugo ToC -->
+
+            <!-- ===!!! Anchor hard wrap-around of Headers 1-6 [see logic above with ($level le 6)] !!!=== -->               
+            <xsl:element expand-text="true" name="a" namespace="http://www.w3.org/1999/xhtml">
+
+               <xsl:attribute name="href">#{@_tree-xml-id}</xsl:attribute>
+               <xsl:attribute name="class">no-anchor-xslt toc{$level} name</xsl:attribute>
+               <xsl:attribute name="title">Focus on {@gi} details</xsl:attribute>
+
+                  <!-- ===!!! The Headers 1-6 that are being wrapped around !!!=== -->
```

Comment:
I suggest we leave it because it gives insight into another "inside of the meat-grinder magic" of Hugo it really helped me to get what's important at the moment and place and what's happening later. So, possibly, it will help someone else.

As to likes/dislikes of element vs HTML tag: element shows "intent in-place". So seeing element's the namespace one can clearly understand intent and namespace mapping. E.g. does it throw XML over to other process/parser/interpreter;  does it reshape the data in the same space? Thus, element is locally narrowing things one has to remember to understand XSL. I also agree that it can be silly with HTML tags - the intent is clearer than water in a glacial lake. Though as a general style choice I'd prefer element stuffing. 

You can call it "Pythonic visual perception" or "white spaces matter for clarity". Element gives clear bigger picture of context. Also, don't underestimate niceness of the clearly aligned attributes stuffing structure under element, while HTML mentality may encourage developers (does not force though) to do attributes packed inside of the angular brackets (seems like "prematurely-optimized" HTML), which could be super long and confusing to understand.